### DISCUSSION TITLE
Getting 419 responses on fresh Jetstream install using ios simulator

### DISCUSSION THREAD

**Discussion by jonnywilliamson** [inertiajs/inertia#336] [Help]
Hi there. I've been stuck with this for hours and need a little help to check that I'm not going insane.

I am developing on my macbook and using the Xcode Simulator to test all my design and code on different iphone models etc. However I keep getting 419 errors ONLY on the simulator with every POST request....When I check in my normal browser everything is perfect.

To make sure it wasn't just me, I have installed a fresh, untouched version of Laravel 8 / jetstream / inertia and I can replicate the same problem. Viewing in browser...no problem. Launching the ios simulator and viewing the site, that's ok. As soon as I post to login/register..BOOM 419 error.

I made a short video showing the entire process. It's **only 1 min** long - it shows

1. Install a new laravel app
2. Install Jetstream
3. Run Npm 
4. Run migration
5. Open local browser and register/login a new user. No problem
6. Open iOS simulatior and visit the site. No problem,
7. Use the ios simulator to register or try and login...419

It's a quick video because I didn't want to bore anyone. But can anyone else recreate this? Is it just me?

I just do not know how or where to debug this. I then installed _pingcrm_ and that DOES work on the ios simulator. Why would a fresh install of jetstream not work?

Any help greatly appreciated.

[![419 Errors on Post](https://user-images.githubusercontent.com/2513663/100557463-4d200680-32a1-11eb-9238-e7452ae09cc7.png)](https://www.youtube.com/watch?v=7FjFfu-Pqtk "419 Errors")

**Comment by maggz69** [comment]
I've experienced this issue before. This error had to do with the cookie not being correctly set on domains when I was accessing pages that needed authentication. This had to do with the sessions config making use of file, I switched to database sessions and haven't experienced this issue since.

**Reply by claudiodekker** [reply]
@jonnywilliamson Could you verify that this indeed solves your answer? 

Based on your issue, at least, I have no idea what's going on, so I'm really hoping that what @maggz69 mentioned is indeed the solution 😅

**Reply by jonnywilliamson** [reply]
This issue took days away from me. 😞 

In the end I ended up trying multiple versions of xcode ios simulator before I finally got one that didn't throw 419 issues.

I don't know if this is an xcode ios simulator bug, or a bug here. All I know is that it was one of the most infurriating things ever!!
